vendor/github.com/weaveworks/go-checkpoint/checkpoint.go

```diff
@@ -284,6 +291,31 @@ func checkResult(r io.Reader) (*CheckResponse, error) {
 	return &result, nil
 }
 
+// getSystemUUID returns the base64 encoded, scrypt hashed contents of
+// /sys/class/dmi/id/product_uuid (with some salt)
+func getSystemUUID() (string, error) {
+	uuid, err := ioutil.ReadFile("/sys/class/dmi/id/product_uuid")
+	if err != nil {
+		return "", err
+	}
+	if len(uuid) <= 0 {
+		return "", fmt.Errorf("Empty system uuid")
+	}
+	hash, err := scrypt.Key(uuid, uuid, 16384, 8, 1, 32)
```

Comment:
> this salt doesn't seem very random

(Random) salt doesn't help here, so this is equivalent to just not using salt.  I should update the comment.

> Might need increasing since 2009?

https://github.com/Tarsnap/scrypt/issues/19